docs/ref-manual/cli/options.md

```diff
@@ -479,4 +484,23 @@ This list is incomplete.
 This option determines whether {ref}`havoc summaries <havoc-summary>` assume
 that the called method returns the correct number of return values.
 
+(-showInternalFunctions)=
+#### `--settings -showInternalFunctions`
+
+This option causes the Prover to output a list of all the potentially
+summarizable internal functions.  In some cases the Prover is unable to locate
+all internal functions, and so summaries may not be applied.  This option can
+be useful to determine whether summary is applied or not.
+
+(-globalTimeout)=
+#### `--settings -globalTimeout=<seconds>`
+
+This option sets the global timeout in seconds.  By default, there is no global
+timeout.
+
+The global timeout is different from the {ref}`--smt_timeout` option: the
+`--smt_timeout` flag constrains the amount of time allocated to the processing
+of each individual rule, while the `-globalTimeout` flag constrains the
+processing of the entire job, including static analysis and other
+preprocessing.
```

Comment:
Worth nothing that this external timeout is enforced somewhat brutally: if we hit it, we try to cancel things reasonably but we will just hard stop any jobs that exceed the global timeout.

Reply:
Here's what I wrote: "Jobs that exceed the global timeout will simply be terminated, so the result
reports may not be generated."  Does that address what's missing?

Reply:
I wasn't aware we have the globaltimeout feature. Is this the 2 hour hard stop?  If so there is a default...
Also are we sure we want to give it to users? I can imagine Silo giving an extremely large number. The whole reason we have the 2 hour hard stop is to save the computing resources

Reply:
I don't believe in security by obscurity.  I think if the feature's there, it should be documented.  If we don't want users using it, we should either remove the option, or use access control if we only want internal users to use it

Reply:
Note to self: I'm running a test to see if this works ([short timeout](https://vaas-stg.certora.com/output/6554/52e5c988e0d35207e737/?anonymousKey=508a15be19e7e55109608ef1fd489a2346c33917), [long timeout](https://vaas-stg.certora.com/output/6554/4814336bd9db1527859f/?anonymousKey=b1ab300f346fb056ab8595836d968fa356f554aa)).  I'll recheck on this when it's done.

Reply:
The short timeout did work, the long timeout didn't run long enough to hit the 2hr mark; I'm trying again with a longer `--smt_timeout`: https://vaas-stg.certora.com/output/6554/467f4209e300b03b850d?anonymousKey=092425fc311689830b985c3f1947767f0ea0ebf3

Reply:
Thanks to @0xDatapunk for running a test; the `--globalTimeout` did not work when set to longer than 2h.  Will update